include/translated-object.php

```diff
@@ -199,38 +115,24 @@ public function get_translations_from_term_id( $term_id ) {
 			return array();
 		}
 
-		// Lang slugs as array keys, template IDs as array values.
+		// Lang slugs as array keys, translation IDs as array values.
 		$translations = maybe_unserialize( $translations_term->description );
 
 		return $this->validate_translations( $translations, 0, 'display' );
 	}
 
 	/**
-	 * Tells whether a translation term must be updated.
-	 *
-	 * @since 2.3
-	 *
-	 * @param int   $id           Object id ( typically a post_id or term_id ).
-	 * @param int[] $translations An associative array of translations with language code as key and translation id as
-	 *                            value. Make sure to sanitize this.
-	 * @return bool
-	 */
-	protected function should_update_translation_group( $id, $translations ) {
-		// Don't do anything if no translations have been added to the group.
-		$old_translations = $this->get_translations( $id ); // Includes at least $id itself.
-		return ! empty( array_diff_assoc( $translations, $old_translations ) );
-	}
-
-	/**
-	 * Saves translations for posts or terms.
+	 * Saves the object's translations.
 	 *
 	 * @since 0.5
 	 *
-	 * @param int   $id           Object id ( typically a post_id or term_id ).
-	 * @param int[] $translations An associative array of translations with language code as key and translation id as value.
-	 * @return int[] An associative array with language codes as key and post ids as values.
+	 * @param int   $id           Object ID.
+	 * @param int[] $translations An associative array of translations with language code as key and translation ID as value.
+	 * @return int[] An associative array with language codes as key and object IDs as values.
+	 *
+	 * @phpstan-return array<non-empty-string, positive-int>
 	 */
-	public function save_translations( $id, $translations ) {
+	public function save_translations( $id, array $translations = array() ) {
 		$id = $this->sanitize_int_id( $id );
```

Comment:
Maybe we can remove it too.

Reply:
If we do that, we must tag `$id` as `positive-int` in the DocBlock, which will trigger lots of issues.